### PR TITLE
Incidents Options exposing

### PR DESCRIPTION
**Pull request by Udumft**
### Description
We need to expose public configuration for Navigator's incidents options. This is required by customers and Android already has such ability.

### Implementation
Added NavigationSettings.liveIncidentsOptions and forwarded this configuration to NavigatorConfig.

### CODE REVIEW

**Comment by Udumft** [comment]
Unfortunately we cannot append new configuration to `ElectronicHorizonOptions` because NavigatorConfig is used only before Navigator initialization, while EH options may be (re)configured at any time. This splits EH configuration into different places which is what I'd want to avoid. Anyway, Android has similar configuration so at least we maintain similar behavior.

File: Sources/MapboxCoreNavigation/IncidentsOptions.swift [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    /**
+     Incidents provider graph name.
+     */
+    public var graph: String
+    /**
+     LTS incidents service API url.
+     
+     If empty line is supplied will use a default url.
+     */
+    public var apiURL: URL
```
Comment (azarovalex) [inline Sources/MapboxCoreNavigation/IncidentsOptions.swift:17]: You cannot create URL with empty line, `URL(string: "")` will return nil.
Let's just use String, this is the parameter NN accepts

File: Sources/MapboxCoreNavigation/NativeHandlersFactory.swift [hunk 2]
```diff
@@ -102,14 +102,21 @@ class NativeHandlersFactory {
     }()
     
     static var navigatorConfig: NavigatorConfig {
+        var nativeIncidentsOptions: MapboxNavigationNative.IncidentsOptions?
+        if let incidentsOptions = NavigationSettings.shared.liveIncidentsOptions,
+           !incidentsOptions.graph.isEmpty,
+           !incidentsOptions.apiURL.absoluteString.isEmpty {
```
Comment (azarovalex) [inline Sources/MapboxCoreNavigation/NativeHandlersFactory.swift:108]: Why do we filter all IncidentsOptions if `apiURL` is an empty string, it's a completely valid case to pass only `graph` parameter.

File: Sources/MapboxCoreNavigation/IncidentsOptions.swift (hunk 1 above)
Comment (Udumft) [reply Sources/MapboxCoreNavigation/IncidentsOptions.swift:17]: Wanted to use URL object for all of it's utility. So probably should allow this one to be nullable instead to maintain functionality...

Comment (azarovalex) [reply Sources/MapboxCoreNavigation/IncidentsOptions.swift:17]: Maybe for parity with Android let's leave it as `String`?
https://docs.mapbox.com/android/navigation/api/2.7.0/libnavigation-base/com.mapbox.navigation.base.options/-incidents-options/

File: Sources/MapboxCoreNavigation/NativeHandlersFactory.swift (hunk 2 above)
Comment (Udumft) [reply Sources/MapboxCoreNavigation/NativeHandlersFactory.swift]: Good catch, I've missed this one

File: Sources/MapboxCoreNavigation/IncidentsOptions.swift (hunk 1 above)
Comment (Udumft) [reply Sources/MapboxCoreNavigation/IncidentsOptions.swift]: We are using `URL` across our SDK for such data. So IMHO it is OK to not mirror Android here

File: Sources/MapboxCoreNavigation/NativeHandlersFactory.swift [hunk 3]
```diff
@@ -102,14 +102,20 @@ class NativeHandlersFactory {
     }()
     
     static var navigatorConfig: NavigatorConfig {
+        var nativeIncidentsOptions: MapboxNavigationNative.IncidentsOptions?
+        if let incidentsOptions = NavigationSettings.shared.liveIncidentsOptions,
+           !incidentsOptions.graph.isEmpty || incidentsOptions.apiURL != nil {
```
Comment (chezzdev) [inline Sources/MapboxCoreNavigation/NativeHandlersFactory.swift:107]: NN has [its own logic](https://github.com/mapbox/mapbox-navigation-native/blob/a8a7dc26f468a39275765ef41357745c6c546a97/src/mapbox/navigation/config.cpp#L178-L189) on these empty fields. It looks like an empty `graph` is always a mistake, but an absent `apiURL` is an expected value.

I think we should either remove this line to allow for states that NN allows (`graph` is empty, but incidents are enabled) and not depend on their implementation changes, or remove `apiURL` check to prohibit invalid case (incidents are on when `graph` is empty, but `apiURL` is set).

**Review by azarovalex — APPROVED** [review]

File: Sources/MapboxCoreNavigation/NativeHandlersFactory.swift (hunk 3 above)
Comment (Udumft) [reply Sources/MapboxCoreNavigation/NativeHandlersFactory.swift:107]: Provided snippet just warns about the empty `graph` value, but still enables the incidents. So I cannot say that it is an error to do so.
This mirrors Android [implementation](https://github.com/mapbox/mapbox-navigation-android/blob/cbb01d24630d444198bd4f19f8435f395d22ad5f/libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt#L313) which is important atm, but we could address this on both platforms as a tailwork.

Comment (chezzdev) [reply Sources/MapboxCoreNavigation/NativeHandlersFactory.swift:107]: Got it. Please create a ticket to sync with Android and make this right, possibly before the release.

**Review by chezzdev — APPROVED** [review]

File: Sources/MapboxCoreNavigation/NavigationSettings.swift [hunk 4]
```diff
@@ -123,12 +125,23 @@ public class NavigationSettings {
...
     }
     
+    /**
+     Configuration on how live incidents on a most probable path are detected.
+     
+     You can override this property by using `NavigationSettings.initialize(directions:tileStoreConfiguration:navigationRouterType:alternativeRouteDetectionStrategy:utilizeSensorData:navigatorPredictionInterval:liveIncidentsOptions:)` method.
+     
+     If set to `nil`, live incidents are turned off (by default).
+     */
+    public var liveIncidentsOptions: IncidentsOptions? {
```
Comment (MaximAlien) [inline Sources/MapboxCoreNavigation/NavigationSettings.swift:141]: If enabled, is this option going to affect billing anyhow? If so, should we mention this?

Comment (Udumft) [reply Sources/MapboxCoreNavigation/NavigationSettings.swift:141]: It should not. Billing we are tracking on platform side is about trips counting, which is not affected by this change. Billing feature-wise is done per user token which is outside of NavSDK scope.

**Pull request merged**